spec/requests/v1/packages_index_spec.rb

```diff
@@ -0,0 +1,55 @@
+# frozen_string_literal: true
+
+require "rails_helper"
+require "checkpoint_helper"
+
+RSpec.describe "GET /v1/packages", :checkpoint_transaction, type: :request do
+  let(:key)  { Keycard::DigestKey.new }
+  let!(:user) { Fabricate(:user, api_key_digest: key.digest) }
+
+  let!(:video1) { Fabricate(:package, content_type: 'video') }
+  let!(:video2) { Fabricate(:package, content_type: 'video') }
+  let!(:audio1) { Fabricate(:package, content_type: 'audio') }
+
+  let(:headers) do
+    {  "Authorization"   => "Token token=#{key}" }
+  end
+
+  context "as a video viewer" do
+    # TODO: extract these granting methods from FakeUser?
+    # TODO: allow specifying something more concise in grant!?
```

Comment:
I've been boggling over whether users/accounts should have role info or a reference to the authority. There are a lot of traps there (cardinality weirdness and authz notions potentially leaking into weird parts of the app). So, I don't know about User getting grant_role! or FakeUser diverging... Maybe.

As far as convenience, we are including the checkpoint helper explicitly, so I'm not really opposed to a couple of utility methods for creating things... but I think we also want that utility elsewhere, so making a couple of value object types seems like a good idea.

Maybe something like this?

```
class Viewer
  def credential_type
    'role'
  end

  def credential_id
    'viewer'
  end
end

module PackageTypes
  def self.any
    Checkpoint::Resource.all
  end

  def self.video
    Checkpoint::Resource::AllOfType.new
  end
end
```